website/source/docs/providers/aws/r/cloudwatch_logs_subscription_filter.html.markdown

```diff
@@ -0,0 +1,39 @@
+---
+layout: "aws"
+page_title: "AWS: aws_cloudwatch_logs_subscription_filter"
+sidebar_current: "docs-aws-resource-cloudwatch-logs-subscription-filter"
+description: |-
+  Provides a CloudWatch Logs subscription filter.
+---
+
+# aws\_cloudwatch\_logs\_subscription\_filter
+
+Provides a CloudWatch Logs subscription filter resource.
+
+## Example Usage
+
+```
+resource "aws_cloudwatch_logs_subscription_filter" "test_lambdafunction_logfilter" {
+  name = "test_lambdafunction_logfilter"
+  role = "${aws_iam_role.iam_for_lambda.arn}"
+  log_group = "/aws/lambda/example_lambda_name"
+  filter_pattern = "logtype test"
+  destination = "${aws_kinesis_stream.test_logstream.arn}"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) A name for the subscription filter
+* `destination` - (Required) The ARN of the destination to deliver matching log events to. Currently only Kinesis stream / a logical destination
+* `filter_pattern` - (Required) A valid CloudWatch Logs filter pattern for subscribing to a filtered stream of log events.
+* `log_group` - (Required) The name of the log group to associate the subscription filter with
+* `role` - (Required) The ARN of an IAM role that grants Amazon CloudWatch Logs permissions to deliver ingested log events to the destination stream
```

Comment:
Role is AFAIK optional - http://docs.aws.amazon.com/cli/latest/reference/logs/put-subscription-filter.html